web/crux-ui/src/elements/dyo-single-file-upload.tsx

```diff
@@ -0,0 +1,47 @@
+import useTranslation from 'next-translate/useTranslation'
+import React from 'react'
+import { DyoButton } from './dyo-button'
+import { DyoLabel } from './dyo-label'
+
+export interface DyoSingleFileUploadProps {
+  name: string
+  label?: string
+  handleFile: Function
+  accept: string
+}
+
+export const DyoSingleFileUploadInput = (props: DyoSingleFileUploadProps) => {
+  const { t } = useTranslation('common')
+  const { label, accept, handleFile } = props
+
+  const hiddenFileInput = React.useRef(null)
+
+  const handleClick = () => {
+    hiddenFileInput.current.click()
+  }
+
+  const handleChange = event => {
+    const fileUploaded = event.target.files[0]
+    if (fileUploaded) {
+      handleFile(fileUploaded)
+    }
+  }
+
+  return (
+    <>
+      {!label ? null : (
+        <DyoLabel className="mt-8 mb-2.5" htmlFor={props.name}>
+          {label}
+        </DyoLabel>
+      )}
+
+      <DyoButton className="px-6 w-28" outlined type="button" onClick={handleClick}>
+        {t('upload')}
+      </DyoButton>
+
+      <input type="file" accept={accept} ref={hiddenFileInput} onChange={handleChange} style={{ display: 'none' }} />
+    </>
+  )
+}
+
+DyoSingleFileUploadInput.displayName = 'DyoSingleFileUpload'
```

Comment:
I think you don't have to define a displayName for the component.

Reply:
That's true, I will remove it, thx